src/hook.ts

```diff
@@ -109,13 +109,13 @@ export function useStorage<T = any>(rawKey: RawKey, onInit?: Setter<T>) {
       if (onInit instanceof Function) {
         const initValue = onInit?.(v, true)
         if (initValue !== undefined) {
-          persistValue(initValue)
+          return persistValue(initValue)
         }
       } else {
         setRenderValue(v !== undefined ? v : onInit)
       }
-      setIsLoading(false)
     })
+    .finally(() => setIsLoading(false))
```

Comment:
Ooh this is very interesting... basically we leverage the end of the promise chain (which is the return of persistValue), then we toggle the loading off!

Reply:
I wonder if this code should be refactored into an await chain, instead of using the finally. This took me a while to understand that the code flow is significantly different from the original code. There's a hidden await on the persistValue

Reply:
Ah yea good point